### PR TITLE
CIV-11219 Wrong page loads on full admit

### PR DESCRIPTION
**Pull request by jeswanth-hmcts**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-11219

### Change description ###

defendant selects full defence, already paid
on claimant response message should be fixed as miro 

 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/routes/features/claimantResponse/settleClaimController.ts [hunk 1]
```diff
@@ -42,12 +43,16 @@ settleClaimController.get(CLAIMANT_RESPONSE_SETTLE_CLAIM_URL, async (req: Reques
...
+    const redisKey = generateRedisKey(req as unknown as AppRequest);
+    const claim: Claim = await getCaseDataFromStore(redisKey);
     const form = new GenericForm(new GenericYesNo(req.body.option, 'ERRORS.VALID_YES_NO_SELECTION'));
     form.validateSync();
     if (form.hasErrors()) {
       renderView(form, res, paidAmount, null);
     } else {
-      await saveClaimantResponse(generateRedisKey(req as unknown as AppRequest), form.model, claimantResponsePropertyName);
+      claim.isFullDefence() && claim.hasPaidInFull() ?
+        await saveClaimantResponse(redisKey, form.model, 'hasFullDefenceStatesPaidClaimSettled') :
```
Comment (laarmada) [inline src/main/routes/features/claimantResponse/settleClaimController.ts:54]: use a constant instead a text

File: src/main/routes/features/claimantResponse/settleClaimController.ts [hunk 2]
```diff
@@ -23,17 +23,18 @@ function renderView(form: GenericForm<GenericYesNo>, res: Response, paidAmount:
...
+    const claim: Claim = await getCaseDataFromStore(generateRedisKey(req));
     let hasPaidInFull: boolean;
     if (claim.isFullDefence()) {
       hasPaidInFull = claim.hasPaidInFull();
       paidAmount = claim.isRejectAllOfClaimAlreadyPaid();
     } else if(claim.isPartialAdmissionPaid()){
       paidAmount = claim.partialAdmissionPaidAmount();
     }
-    renderView(new GenericForm(claim.claimantResponse?.hasPartPaymentBeenAccepted), res, paidAmount, hasPaidInFull);
+    const isClaimSettled = claim.isFullDefence() && claim.hasPaidInFull() ? claim.claimantResponse?.hasFullDefenceStatesPaidClaimSettled : claim.claimantResponse?.hasPartPaymentBeenAccepted;
```
Comment (laarmada) [inline src/main/routes/features/claimantResponse/settleClaimController.ts:36]: from 29 until 36 is logic that should be in the service class.... only my opinion ... something like:

function getSettleClaim(){}

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/yourResponseSectionTasks.ts [hunk 3]
```diff
@@ -24,15 +24,16 @@ export function getHaveYouBeenPaidTask(claim: Claim, claimId: string, lang: stri
...
-    description: t('CLAIMANT_RESPONSE_TASK_LIST.YOUR_RESPONSE.SETTLE_CLAIM_FOR', {
+    description: t(text, {
       lng: lang,
       paidAmount: claim.isFullDefence() ? claim.isRejectAllOfClaimAlreadyPaid() : claim.partialAdmissionPaidAmount(),
     }),
     url: constructResponseUrlWithIdParams(claimId, CLAIMANT_RESPONSE_SETTLE_CLAIM_URL),
     status: TaskStatus.INCOMPLETE,
   };
-  if (claim?.hasClaimantSettleTheClaimForDefendantPartlyPaidAmount() ||
+  if (claim?.hasClaimantSettleTheClaimForDefendantPartlyPaidAmount() || claim.claimantResponse?.hasFullDefenceStatesPaidClaimSettled?.option ||
```
Comment (laarmada) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/yourResponseSectionTasks.ts:36]: remove the unnecessaries ? in claim?

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts [hunk 4]
```diff
@@ -109,18 +109,22 @@ export function buildWhatToDoNextSection(claim: Claim, claimId: string, lang: st
...
-  if (claim.isPartialAdmissionPaid() || claim.responseStatus === ClaimResponseStatus.RC_PAID_LESS) {
-    if(claim.hasClaimantConfirmedDefendantPaid()){
-      tasks.push(getSettleTheClaimForTask(claim, claimId, lang));
-    }
-    if(claim.hasClaimantRejectedDefendantPaid() || claim.hasClaimantRejectedPartAdmitPayment()){
-      tasks.push(getFreeTelephoneMediationTask(claim, claimId, lang));
-    }
+  const isFullDefencePaidTotal = claim.isFullDefence() && claim.hasPaidInFull();
+  const isRespondentPartialPaid = (claim.isPartialAdmissionPaid() || claim.responseStatus === ClaimResponseStatus.RC_PAID_LESS);
+  const isSettleTheClaimTaskEnable = (isRespondentPartialPaid && claim.hasClaimantConfirmedDefendantPaid()) || isFullDefencePaidTotal;
```
Comment (laarmada) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts:114]: try to cut the vbles names are so long...

Comment (laarmada) [reply src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts:114]: isFullDefencePaidTotal 
 isRespondentPartialPaid
 
we are under claimantResponse journey so defendant/respodent is redundant please use something like:

isPartialPaid
isFullPaid

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts [hunk 5]
```diff
@@ -109,18 +109,22 @@ export function buildWhatToDoNextSection(claim: Claim, claimId: string, lang: st
...
+  const isFreeTelePhoneMediationTaskEnable = (isRespondentPartialPaid && (claim.hasClaimantRejectedDefendantPaid() || claim.hasClaimantRejectedPartAdmitPayment())) || claim.hasClaimantRejectedDefendantResponse();
```
Comment (laarmada) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts:115]: is redundant TaskEnable, from my point of view:

isFreePhoneMediation

we are under TaskListService folder so task is redundant an also is with enable .....

**Review by laarmada — CHANGES_REQUESTED** [review]

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts (hunk 4 above)
Comment (jeswanth-hmcts) [reply src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasklistBuilder.ts]: shorted the names to 'isPartialPaid'
'isFullPaid'

File: src/main/routes/features/claimantResponse/settleClaimController.ts (hunk 1 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/claimantResponse/settleClaimController.ts]: moved to constants

**Review by laarmada — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Comment by jarekPierchala** [comment]
Will be part of post-case-progression-tasks branch

**Pull request closed without merging**